src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -715,26 +715,37 @@ private BoundExpression RebindSimpleBinaryOperatorAsConverted(BoundBinaryOperato
             var stack = ArrayBuilder<BoundBinaryOperator>.GetInstance();
             BoundBinaryOperator? current = unconvertedBinaryOperator;
 
-            while (current != null)
+            var result = doRebind(diagnostics, stack, current);
+            stack.Free();
+            return result;
+
+            BoundExpression doRebind(BindingDiagnosticBag diagnostics, ArrayBuilder<BoundBinaryOperator> stack, BoundBinaryOperator? current)
             {
-                Debug.Assert(current.IsUnconvertedInterpolatedStringAddition);
-                stack.Push(current);
-                current = current.Left as BoundBinaryOperator;
-            }
+                while (current != null)
+                {
+                    Debug.Assert(current.IsUnconvertedInterpolatedStringAddition);
+                    stack.Push(current);
+                    current = current.Left as BoundBinaryOperator;
+                }
 
-            Debug.Assert(stack.Count > 0 && stack.Peek().Left is BoundUnconvertedInterpolatedString);
+                Debug.Assert(stack.Count > 0 && stack.Peek().Left is BoundUnconvertedInterpolatedString);
 
-            BoundExpression? left = null;
-            while (stack.TryPop(out current))
-            {
-                Debug.Assert(current.Right is BoundUnconvertedInterpolatedString);
-                left = BindSimpleBinaryOperator((BinaryExpressionSyntax)current.Syntax, diagnostics, left ?? current.Left, current.Right, leaveUnconvertedIfInterpolatedString: false);
-            }
+                BoundExpression? left = null;
+                while (stack.TryPop(out current))
+                {
+                    var right = current.Right switch
+                    {
+                        BoundUnconvertedInterpolatedString s => s,
+                        BoundBinaryOperator b => doRebind(diagnostics, stack, b),
```

Comment:
>doRebind [](http://example.com/codeflow?start=49&length=8)

It looks like this call is not going to return until it purges the entire stack, which contains entries for parent nodes. When that happens the call doesn't return the node that corresponds to the ```current.Right```, but a node that corresponds to its parent, or quite possibly some node that doesn't correspond to any original node. It doesn't feel like desired behavior.